kiteconnect/ticker.py

```diff
@@ -624,22 +621,10 @@ def set_mode(self, mode, instrument_tokens):
 
     def resubscribe(self):
         """Resubscribe to all current subscribed tokens."""
-        modes = {}
-
-        for token in self.subscribed_tokens:
-            m = self.subscribed_tokens[token]
-
-            if not modes.get(m):
-                modes[m] = []
-
-            modes[m].append(token)
-
-        for mode in modes:
-            if self.debug:
-                log.debug("Resubscribe and set mode: {} - {}".format(mode, modes[mode]))
+        log.debug("Resubscribe and set mode: {} - {}".format(self.subscribe_tokens, self.subscribe_mode))
 
-            self.subscribe(modes[mode])
-            self.set_mode(mode, modes[mode])
+        self.subscribe(self.subscribe_tokens)
+        self.set_mode(self.subscribe_mode, self.subscribe_tokens)
```

Comment:
What is `subscribe_mode` here? and you know that each subscription can have different modes right? The line above subscribes to all the tokens which default to quote mode but what about the modes which got changed via `setMode`? This is why we maintain a map that has token as key and corresponding mode as value.